### ISSUE TITLE
refactor _get_sftp_backend of the testsuite to use env vars

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borgstore#42]
Currently, username, hostname, port of a private testing host of mine is configured there.

That should be generalized so everybody can run the sftp tests, just read the required values from some env vars.

**Comment by ThomasWaldmann** [comment]
@mirko ^ can you do a PR for that while testing #38?

**Comment by ThomasWaldmann** [comment]
You could do the testing also by running the test suite IF that stuff was not hard coded.

Guess it just needs one env var like `BORGSTORE_SFTP_TEST_URL=...` and then call `get_sftp_backend()` with that (and remove all hardcoded stuff).

**Comment by mirko** [comment]
I wasn't immediately able to run the test suite - where is `key` to be (from) referenced here [https://github.com/borgbackup/borgstore/blob/90b4363c1f1262fc1fc7c6120f2ce114876ae797/tests/test_backends.py#L9]?

Anyway, without having been able to test this, this is probably how I'd start doing so:
```
diff --git a/tests/test_backends.py b/tests/test_backends.py
index 9a15a35..3d729c5 100644
--- a/tests/test_backends.py
+++ b/tests/test_backends.py
@@ -3,6 +3,7 @@ Generic testing for the misc. backend implementations.
 """
 
 from pathlib import Path
+from os import environ
 
 import pytest
 
@@ -31,29 +32,29 @@ def posixfs_backend_created(tmp_path):
         be.destroy()
 
 
-def _get_sftp_backend(local=False, remote_url=False, remote_config=True):
+def _get_sftp_backend(host, dst_url, user=None, port=0):
     # needs an authorized key loaded into the ssh agent. pytest works, tox doesn't:
-    if local:
-        return Sftp(username="tw", hostname="localhost", path="/Users/tw/w/borgstore/temp-store")
-    # for tests with higher latency:
-    if remote_url:
-        return Sftp(
-            username="twaldmann",
-            hostname="shell.ipv4.thinkmo.de",
-            port=2222,
-            path="/home/twaldmann/borgstore/temp-store",
-        )
-    # same as previous, but loads config for host "shell" from ~/.ssh/config:
-    if remote_config:
-        return Sftp(hostname="shell", path="/home/twaldmann/borgstore/temp-store")
+    return Sftp(
+        hostname = host,
+        username = user, # optional, fine to be `None`
+        port     = port and port or 0, # optional, 0 defaults to default port
+        path     = dst_url,
+    )
 
     raise ValueError("check _get_sftp_backend() parameter defaults!")
 
 
 def check_sftp_available():
     """in some test environments, get_sftp_backend() does not result in a working sftp backend"""
+    if 'BORGSTORE_BACKEND_SFTP_HOST' not in environ:
+        raise ValueError("Environment variable `BORGSTORE_BACKEND_SFTP_HOST` required to be set")
     try:
-        be = _get_sftp_backend()
+        be = _get_sftp_backend(
+                hostname = environ['BORGSTORE_BACKEND_SFTP_HOST'],
+                username = 'BORGSTORE_BACKEND_SFTP_USER' in environ and environ['BORGSTORE_BACKEND_SFTP_USER'],
+                port     = 'BORGSTORE_BACKEND_SFTP_PORT' in environ and environ['BORGSTORE_BACKEND_SFTP_PORT'] or 0,
+                path     = 'BORGSTORE_BACKEND_SFTP_PATH' in environ and environ['BORGSTORE_BACKEND_SFTP_PATH'] or '/',
+                )
         be.create()  # first sftp activity happens here
     except Exception:
         return False
```

**Comment by ThomasWaldmann** [comment]
`key` is imported from `tests/__init__.py`. why do you ask?

**Comment by ThomasWaldmann** [comment]
I'll give it a shot myself. :-)

**Comment by mirko** [comment]
> I'll give it a shot myself. :-)

Probably for the better :) - maybe you can (re)use some pieces of above posted diff

**Comment by ThomasWaldmann** [comment]
I usually run `pytest -v -rs` from the toplevel dir.

I pushed an implementation to the PR.

**Comment by ThomasWaldmann** [comment]
Fixed by #38.
